### PR TITLE
Fix memory corruption bug

### PR DESCRIPTION
**Pull request by kkourt**
This PR includes a bug that fixes a memory corruption bug, and it includes some other minor patches as well. Please review patch-by-patch.

Fixes: https://github.com/cilium/tetragon/issues/1083
Fixes: https://github.com/cilium/tetragon/issues/193

### CODE REVIEW

File: pkg/observer/observer_stats.go [hunk 1]
```diff
@@ -41,44 +41,49 @@ func (s *statValue) GetValuePtr() unsafe.Pointer {
...
-				continue
-			}
-			sum := int64(0)
-			for cpu := int(0); cpu < runtime.NumCPU(); cpu++ {
-				sum += v.Value[cpu]
-			}
-			mapmetrics.MapSizeSet(m.Name, int(mapLink.MapInfo.MaxEntries), float64(sum))
-			mapLink.Close()
-			mapLinkStats.Close()
+	v, ok := value.DeepCopyMapValue().(*statValue)
```
Comment (kevsecurity) [inline pkg/observer/observer_stats.go:71]: Do we need the deep copy here? It looks like we just sum the values. Could we cast instead and save a copy?

**Review by kevsecurity — APPROVED** [review]
LGTM. One comment unrelated, so feel free to ignore for now.

File: pkg/observer/observer_stats.go (hunk 1 above)
Comment (kkourt) [reply pkg/observer/observer_stats.go:71]: I think you are right.
The deepcopy was in the original code though, and maybe this is worth doing as a followup.

File: pkg/bpf/map_linux.go [hunk 2]
```diff
@@ -158,14 +158,17 @@ func OpenMap(name string) (*Map, error) {
...
 		return nil, err
 	}
 
 	if info.MapType == 0 {
+		unix.Close(fd)
 		return nil, fmt.Errorf("Unable to determine map type")
 	}
 
 	if info.KeySize == 0 {
+		unix.Close(fd)
```
Comment (mtardy) [inline pkg/bpf/map_linux.go:171]: I was wondering if there was a reason why we can't use `defer(unix.Close(fd))` here after `ObjGet(name)`?

Comment (mtardy) [reply pkg/bpf/map_linux.go:171]: aaaah yeah sorry indeed!

**Pull request merged**